lib/miq_automation_engine/engine/miq_ae_event.rb

```diff
@@ -128,4 +130,14 @@ def self.automate_user
   end
 
   private_class_method :automate_user
+
+  def self.event_tenant(event)
+    case event.class
+    when EmsEvent
+      event.ext_management_system.tenant
+    when MiqEvent
+      (event.target.ext_management_system || event.target).tenant
```

Comment:
@lfu do all targets respond to `ext_management_system`?
For foreman, do we want manager here instead?